lib/ansible/modules/cloud/azure/azure_rm_servers_facts.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_servers
+version_added: "2.5"
+short_description: Get Servers facts.
+description:
+    - Get facts of Servers.
+
+options:
+    resource_group_name:
+        description:
+            - The name of the resource group that contains the resource. You can obtain this value from the Azure Resource Manager API or the portal.
+        required: True
+    server_name:
+        description:
+            - The name of the server.
+        required: False
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+      - name: Get instance of Servers
+        azure_rm_servers:
+          resource_group_name: "{{ resource_group_name }}"
+          server_name: "{{ server_name }}"
+
+      - name: List instances of Servers
+        azure_rm_servers:
+          resource_group_name: "{{ resource_group_name }}"
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.sql import SqlManagementClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMServersFacts(AzureRMModuleBase):
+    def __init__(self):
+        # define user inputs into argument
+        self.module_arg_spec = dict(
+            resource_group_name=dict(
+                type='str',
+                required=True
+            ),
+            server_name=dict(
+                type='str',
+                required=False
+            ),
+        )
+        # store the results of the module operation
+        self.results = dict(
+            changed=False,
+            ansible_facts=dict(azure_dnsrecordset=[])
+        )
+        self.resource_group_name = None
+        self.server_name = None
+        super(AzureRMServersFacts, self).__init__(self.module_arg_spec)
+
+    def exec_module(self, **kwargs):
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        if (self.resource_group_name is not None and
+                self.server_name is not None):
+            self.results['ansible_facts']['get'] = self.get()
+        elif (self.resource_group_name is not None):
```

Comment:
brackets is no need for conditional check

Reply:
this is just autogeneration stuff, works in the same way for one or more items. Brackets don't break sanity tests.
But i can improve this anyway.